docs/decisions/0001-launch-complete-course.rst

```diff
@@ -0,0 +1,421 @@
+   :width: 653
+
+Cons
+----
+
+- We will need to add functionality to validate the requested paths
+  for valid LTI launch paths, so we can restrict LTI launch users' views
+  that should not be seen by them on the MFE.
+- If this is not properly implemented it could lead to [Broken Access Control](https://owasp.org/Top10/A01_2021-Broken_Access_Control/) issues.
+- We could lose our implementation on future learning MFE changes.
+
+Pros
+----
+
+- We can add the functionality we require without adding a new MFE.
+- In the case we propose these changes to the community, it will be easier to merge.
+- We have control over what to render on our view without the need of creating
+  a new MFE or creating new components to render course content from scratch.
+- We have a path to render the course content on the learning MFE that is
+  different from a normal user, having a more clear separation of the
+  LTI launch paths from normal course content learning MFE paths.
+
+Approach 4: Render using a view on our plugin
+=============================================
+
+The learning MFE renders the content of a course using the render_xblock view,
+this view can be either used as a return of another view or use its
+URL on the LMS to render course content. We can use this endpoint to
+render an XBlock on an iframe inside a view for LTI launches on a whole course,
+alongside a navigation menu we could create to let the user navigate across
+the course on our view inside the plugin. Another method can be using the
+unpublished API xblock_view to inject the XBlock HTML into our template,
+we can get a reference on how this works by using the LTI 1.3 content libraries
+implementation. This method will require us to go through the course sequences
+ourselves using the course-home API to create our course navigation menu
+and course home outline. We can use the Paragon library to make our menus
+look the same as the learning MFE. This approach will require the LTI plugin.
+
+Workflow
+--------
+
+1. A new view will receive the IDs of the requested content.
+2. We will validate that the requested IDs are valid and that the user on the request
+   have an LTIPRofile instance.
+3. If the user has no permission, then the view returns nothing.
+4. If the user has permission to load the view, we will call the render_xblock view
+   to render the requested XBlock to the user alongside a navigation menu with
+   the course outline. (This could be done by either injecting the XBlock using
+   the xblock_view unpublished API to inject the HTML fragment or load it with
+   the render_xblock view has an Iframe, similar to how the MFE renders
+   course content).
```

Comment:
On what level of the course hierarchy do you plan to call render_xblock? The unit, I assume?

Reply:
Thanks @kdmccormick!, I will try to modify or extend what was proposed in approach 3 but with a more agnostic approach that could work for this feature. Right now I'm not sure about any performance issues, the performance issue of the iframe should be similar to the learning MFE courseware since it's the same method the learning MFE uses to render a unit content.

Our plan is to call render_xblock on the unit level and add a template to render the sequence navigation on top, similar to how it's done on the learning MFE, currently, you can call render_xblock on a sequence, but renders a sequence navigation that has a broken "Previous" button, there is a change that, if this could be fixed on the edx-platform side, we could render the content at a sequence level and leverage the rendering of the sequence navigation to render_xblock, I still haven't found what generates this issue.

Reply:
> Our plan is to call render_xblock on the unit level and add a template to render the sequence navigation on top, similar to how it's done on the learning MFE

Perfect, I think that is the right way to go 👍🏻  The fact that you can render_xblock can be called with a sequence at all is a legacy behavior that probably should have been removed when we fully switched over to the Learning MFE. ([here's some extra context on why we don't want to directly render xblocks above the unit level, if you're curious](https://github.com/openedx/edx-platform/blob/master/docs/decisions/0006-role-of-xblock.rst#decision)).

Reply:
It wasn't still clear to me if that sequence navigation was intended or not, since this is considered legacy I will not look further into fixing it. This context also will be useful to understanding the role of an XBlock and how we should work with it when we get to add LTI grading or other services. Thank you for sharing this ADR and giving us feedback on this!

Reply:
No problem!